Dark Mode

jduong09 opened the pull request:
Resolves #38 

## Changes/Additions
- Installed 'next-themes'
- Added Dark Mode Button
- Designed Dark Mode color scheme

## How To Test
- Checkout to `issue-38--ui-dark-mode` and pull changes with `git pull`
- Run `npm run dev` to open `localhost:3000`
- Test Dark Mode button, and play game
- Review color scheme

jduong09 commented:
## Update 6/17/25

- Designed Dark Mode Button to be toggle button with Sun and Moon icons, visually representing each mode.

ziggysauce commented on app/page.js:
```diff
@@ -1,14 +1,24 @@
+    setMounted(true)
+  }, []);
+
+  if (!mounted) {
+    return false;
+  }
```
why is this necesasry?

jduong09 replied:
This is added to avoid a hydration mismatch error when using the 'next-themes' package.

ziggysauce commented on app/page.module.css:
```diff
@@ -1,23 +1,9 @@
   font-weight: 700;
   font-size: 1.15rem;
-  color: #E43D12;
+  /*color: #E43D12; */
```
remove the commented out code if not used. or if you need to keep them here for reference, add a comment as to why

ziggysauce commented on components/game/game.module.scss:
```diff
@@ -1,25 +1,26 @@
+  }
+}
+
+.shuffleBtn:hover {
```
keep in mind the hover effect does not apply to mobile devices

ziggysauce commented on components/header/header.js:
```diff
@@ -1,12 +1,33 @@
 import React, { useState } from 'react';
-import pageStyles from '../app/page.module.css';
+import PropTypes from "prop-types";
+import Image from 'next/image';
```
remember my previous comment about mixing single and double quotes. pick one an d stay consistent with it within each file

ziggysauce reviewed (changes requested):
excellent stuff here!

when zooming in real close, you can tell the "Rules" header and the theme toggle are not vertically aligned:

<img width="492" alt="Screenshot 2025-06-24 at 8 50 21 AM" src="https://github.com/user-attachments/assets/7bf4c88b-03fc-4cbe-ac2b-c27ea1b1bc9b" />

this is what it _should_ look like when it's vertically aligned:

<img width="471" alt="Screenshot 2025-06-24 at 8 50 41 AM" src="https://github.com/user-attachments/assets/f220df3c-4678-400b-b4f3-e0074ac5aebe" />

^^ kinda hard to tell, but if you highlight over the elements in your dev tools you'll see it. simply adding some flexbox alignment styles fixes this

---

From far away (and maybe even on mobile) the sun icon looks a bit more like a gear icon, which may confuse the user into thinking it's a "settings" toggle instead of a theme toggle

<img width="794" alt="Screenshot 2025-06-24 at 8 52 05 AM" src="https://github.com/user-attachments/assets/86bbb3d1-251f-43a6-9013-e733835c8444" />

I don't necessarily think you need to change the icon, but perhaps change some of the text or background colors and play around with it to see what looks best. perhaps also make the toggle a little bit bigger so that the height of the toggle matches the "Rules" text. That makes the your app look more uniform in general.

---

Dark mode looks sick. However, I will say the contrast is a bit tough making it difficult to see the cards and the "your hand" box. I would suggest either changing the card background to a lighter color in dark mode (i know you have a feature request to change card backs in general, but this could help at the start simply for dark mode) or changing the background color to not be as dark.

I do like that your _darkened_ the white of the cards in dark mode. at first i thought it would make more sense to just invert and use a "dark" card with lighter colored text for the numbers and suites, but i think the slightly darker white/grey works well here.

ziggysauce commented:
also side note, in your PR testing instructions, you should include that the user should run `npm install` or `npm i` if there are new dependencies. it's just common practice and you're best getting used to it in your PR writeups.

jduong09 commented:
## Update 7/8/25
- Changed sun icon for dark mode toggle
- Aligned dark mode toggle with other list items
- Changed background of card face down for better contrast in dark mode
### Cleanup
- Removed unnecessary comments
- Changed all import statements to use double quotes for consistency

ziggysauce reviewed (changes requested):
ayee changes look good. honestly i'd make the "Rules" button and the dark/light toggle bigger. They can match the height/size of the "Thirteen" header imo.

i'd also make the entire toggle clickable and not just the icon. it just makes for a better user experience not having to hunt for the tap target.

new card backs look fun!

jduong09 commented:
## Update 7/14/25

- Resolved merge conflicts from other branch changes
- Updated PR to have rules and dark mode toggle to match font size as H1.

Merging Pull Request.
